brave/brave-core
# Support wallet_watchAsset

Resolves https://github.com/brave/brave-browser/issues/17878

Image parameter is not supported yet because we currently don't support getting a logo from a remote URL, currently it will still use the placeholder icon if a remote URL is passed. (https://github.com/brave/brave-browser/issues/20000)

## Submitter Checklist:


## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:
1. Visit https://www.coingecko.com/en/coins/railgun and click on the fox icon to add suggest token
<img width="1163" alt="add_to_metamask" src="https://user-images.githubusercontent.com/4730197/145308959-61b8add5-8a79-48b5-820d-e61a15a09699.png">

2. Prompt should show up for adding suggest token (will be asking to unlock first if wallet is locked)
<img width="318" alt="add_suggest_token" src="https://user-images.githubusercontent.com/4730197/145309035-61ed67a0-0620-46e7-a8e7-8c99c372b650.png">

3. Cancel the request.
4. Open a new tab and visit brave://wallet/crypto/portfolio should not see the new token.
5. Do step 1 again and accept the request this time.
6. Go to brave://wallet/crypto/portfolio should see it being listed in visible assets. (Existing page will need to fresh to see the change)
<img width="1144" alt="assets" src="https://user-images.githubusercontent.com/4730197/145309145-26963d84-60ad-4540-98d7-5452eb2a53d8.png">

7. Visit https://www.coingecko.com/en/coins/tether and click on the fox icon to add suggest token
8. Prompt should show up with correct logo.
<img width="321" alt="usdt" src="https://user-images.githubusercontent.com/4730197/145317959-e28a687c-5e79-4efc-91be-c15ad591fab6.png">

 9. Accept the request.
 10. Open a new tab and visit brave://wallet/crypto/portfolio should see tether in the asset list.
<img width="1100" alt="Screen Shot 2021-12-08 at 5 33 17 PM" src="https://user-images.githubusercontent.com/4730197/145318211-84e676c4-979a-41a3-bcf2-245ac9ec12b7.png">
